arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -146,82 +130,54 @@ void DatabaseManagerThread::run() {
       }
 
       if (database != nullptr) {
-        if (!ServerState::instance()->isCoordinator()) {
-          TRI_ASSERT(!database->isSystem());
-
-          if (dealer.isEnabled()) {
-            // remove apps directory for database
-            std::string const& appPath = dealer.appPath();
-            if (database->isOwnAppsDirectory() && !appPath.empty()) {
-              std::lock_guard lockCreate{databaseFeature._databaseCreateLock};
-
-              // but only if nobody re-created a database with the same name!
-              std::lock_guard lock{databaseFeature._databasesMutex};
-
-              TRI_vocbase_t* newInstance =
-                  databaseFeature.lookupDatabase(database->name());
-              TRI_ASSERT(newInstance == nullptr ||
-                         newInstance->id() != database->id());
-              if (newInstance == nullptr) {
-                std::string const dirName = ::getDatabaseDirName(
-                    database->name(), std::to_string(database->id()));
-                std::string path = arangodb::basics::FileUtils::buildFilename(
-                    arangodb::basics::FileUtils::buildFilename(appPath, "_db"),
-                    dirName);
-
-                if (TRI_IsDirectory(path.c_str())) {
-                  LOG_TOPIC("041b1", TRACE, arangodb::Logger::FIXME)
-                      << "removing app directory '" << path << "' of database '"
-                      << database->name() << "'";
-
-                  TRI_RemoveDirectory(path.c_str());
-                }
-              }
-            }
-          }
+        TRI_ASSERT(!database->isSystem());
+        if (ServerState::instance()->isCoordinator()) {
```

Comment:
The only thing the thread will do on coordinators is to physically remove the database objects that are in `_droppedDatabases` and that are not marked as "in use" anymore.

Reply:
Understand. What about else (if we don't find anything on _droppedDatabases) branch? Is it necessary on coordinator.

Also what do you think about not start this thread on coordinator, and don't push anything to droppedDatabases.
Instead just call dtor + dealloc when we drop it

Reply:
Ideally, this thread should not be necessary at all on coordinators.
If we drop database objects immediately, it has a potential side-effect though: databases that are still in use elsewhere will then be deleted more quickly. So we may run into segfaults quickly if a database was looked up using `lookupDatabase(...)` and there is not protection (`useDatabase(...)`) in place. 
Previously, due to the potential delay that the DatabaseManagerThread has added post-deletion, these problems may have been hidden. I am not saying it's good to hide problems, just saying here that the change could lead to more crashes unless the other (potential) issues are fixed.

Reply:
>If we drop database objects immediately, it has a potential side-effect though

Yes you're right, we can dealloc only dangling database.
Callback on release can be solution, but I'm not insist.
And I don't do it in this PR

> So we may run into segfaults quickly if a database was looked up using lookupDatabase(...) and there is not protection (useDatabase(...)) in place.

But it's another issue, and probably should be fixed

Reply:
What about else (if we don't find anything on _droppedDatabases) branch? Is it necessary on coordinator?

This question still actual

Reply:
Yes, the only code that the coordinator will execute is dropped all dangling databases in the `_droppedDatabases` list.
Everything else is skipped anyway on the coordinator.